data/json/recipes/recipe_ammo.json

```diff
@@ -27,6 +27,27 @@
   ]
 },{
   "type" : "recipe",
+  "result": "40mm_toxicgas",
+  "category": "CC_AMMO",
+  "subcategory": "CSC_AMMO_BULLETS",
+  "skill_used": "fabrication",
+  "difficulty": 3,
+  "time": 100,
+  "reversible": false,
+  "autolearn": false,
+  "book_learn": [[ "textbook_gaswarfare", 4, "" ] , [ "manual_launcher", 4 ]],
```

Comment:
This feature is actually documented, see do/JSON_INFO.md (but as we all know, nobody reads the documentation):

> "book_learn": [              // (optional) Array of books that this recipe can be learned from. Each entry contains the id of the book and the skill level at which it can be learned.
>    [ "textbook_anarch", 7, "something" ], // The optional third entry defines a name for the recipe as it should appear in the books description (default is the name of resulting item of the recipe)
>    [ "textbook_gaswarfare", 8, "" ] // If the name is empty, the recipe is hidden, it will not be shown in the description of the book.
> ],